rpcs3/Emu/RSX/VK/vkutils/sampler.cpp

```diff
@@ -52,4 +102,114 @@ namespace vk
 
 		return true;
 	}
+
+	sampler_pool_key_t sampler_pool_t::compute_storage_key(
+		VkSamplerAddressMode clamp_u, VkSamplerAddressMode clamp_v, VkSamplerAddressMode clamp_w,
+		VkBool32 unnormalized_coordinates, float mipLodBias, float max_anisotropy, float min_lod, float max_lod,
+		VkFilter min_filter, VkFilter mag_filter, VkSamplerMipmapMode mipmap_mode, const vk::border_color_t& border_color,
+		VkBool32 depth_compare, VkCompareOp depth_compare_mode)
+	{
+		sampler_pool_key_t key{};
+
+		bool use_border_encoding = false;
+		if (border_color.value > VK_BORDER_COLOR_INT_OPAQUE_WHITE)
+		{
+			// If there is no clamp to border in use, we can ignore the border color entirely
+			if (clamp_u == VK_SAMPLER_ADDRESS_MODE_CLAMP_TO_BORDER ||
+				clamp_v == VK_SAMPLER_ADDRESS_MODE_CLAMP_TO_BORDER ||
+				clamp_w == VK_SAMPLER_ADDRESS_MODE_CLAMP_TO_BORDER)
+			{
+				use_border_encoding = true;
+			}
+		}
+
+		key.base_key = u16(clamp_u) | u64(clamp_v) << 3 | u64(clamp_w) << 6;
+		key.base_key |= u64(unnormalized_coordinates) << 9;            // 1 bit
+		key.base_key |= u64(min_filter) << 10 | u64(mag_filter) << 11; // 1 bit each
+		key.base_key |= u64(mipmap_mode) << 12;   // 1 bit
+
+		if (!use_border_encoding)
```

Comment:
can be reversed

Reply:
The default is to not use border encoding. It is more readable this way.